src/blocks/HeaderBlockBrandPivot/MenuItem.tsx

```diff
@@ -127,100 +157,137 @@ interface Effects {
-    initialState={{
-      isOpen: false,
-      isClosing: false,
-      closeAnimationTimeout: undefined,
-      closeTimeout: undefined,
-    }}
-    effects={{
-      open: () => ({ setState, state }) => {
-        if (state.closeTimeout) {
-          window.clearTimeout(state.closeTimeout)
-        }
-        if (state.closeAnimationTimeout) {
-          window.clearTimeout(state.closeAnimationTimeout)
-        }
-        setState({
-          isOpen: true,
-          isClosing: false,
-          closeTimeout: undefined,
-          closeAnimationTimeout: undefined,
-        })
-      },
-      close: () => ({ setState }) => {
-        setState({
-          closeTimeout: window.setTimeout(() => {
-            setState({
-              isOpen: false,
-              isClosing: false,
-              closeTimeout: undefined,
-              closeAnimationTimeout: undefined,
-            })
-          }, 500),
-          closeAnimationTimeout: window.setTimeout(() => {
-            setState({ isClosing: true })
-          }, 250),
-        })
-      },
-      closeWithoutDelay: () => ({ setState }) => {
-        setState({
-          isClosing: true,
-          closeTimeout: window.setTimeout(() => {
-            setState({
-              isClosing: false,
-              isOpen: false,
-              closeTimeout: undefined,
-            })
-          }, 250),
-        })
-      },
-    }}
-  >
-    {({ isOpen, isClosing, open, close, closeWithoutDelay }) => (
-      <MenuListItem
-        key={menuItem._uid}
-        onMouseOver={() => {
-          open()
-        }}
-        onMouseOut={() => close()}
-      >
-        {menuItem.link && menuItem.link.cached_url ? (
-          <MenuLink href={getStoryblokLinkUrl(menuItem.link)}>
-            {menuItem.label}
-          </MenuLink>
-        ) : (
-          <MenuFakeLink>{menuItem.label}</MenuFakeLink>
-        )}
-
-        {menuItem.menu_items && menuItem.menu_items.length > 0 && (
-          <Toggler onClick={isOpen ? closeWithoutDelay : open} isOpen={isOpen}>
-            <Chevron />
-          </Toggler>
-        )}
-
-        {menuItem.menu_items && menuItem.menu_items.length > 0 && (
-          <AnimateHeight height={isOpen && !isClosing ? 'auto' : 0}>
-            <DropdownMenuItemList isOpen={isOpen} isClosing={isClosing}>
-              {menuItem.menu_items.map((innerMenuItem) =>
-                innerMenuItem.link?.cached_url ? (
-                  <MenuListItem key={innerMenuItem._uid}>
-                    <DropdownMenuLink
-                      href={getStoryblokLinkUrl(innerMenuItem.link)}
-                    >
-                      {innerMenuItem.label}
-                    </DropdownMenuLink>
-                  </MenuListItem>
-                ) : null,
-              )}
-            </DropdownMenuItemList>
-          </AnimateHeight>
-        )}
-      </MenuListItem>
+export const SubMenuLinks: React.FC<{
+  menu_items: readonly MenuItemType[]
```

Comment:
We're not so strict on readonly types but doesn't hurt to add it

Reply:
Good to know. In this case I had to. Since the original array is readonly, I can't pass a readonly to non-read only